### ISSUE TITLE
Unexpected slow-down with `@jit` on simple functions that only use element-wise operations and `jnp.roll()` on CPUs

### ISSUE THREAD

**Issue by pmocz** [openxla/xla#18478]
See issue raised with JAX: https://github.com/jax-ml/jax/issues/24373

Simple colab reproducer: https://colab.research.google.com/drive/1pnnC3-XY8-K0MvUv9gH_on2-USG1ZaeY?usp=sharing

Unexpected major slowdown particular to the XLA:CPU compiler is seen when using `@jax.jit` on a fairly simple function that only use element-wise operations and `jnp.roll()` .  Works well running on TPUs

**Comment by penumallas1261** [comment]
Hi @pmocz ,

Can you pleas try with latest version? I tried with jax 0.8.0 version and I cannot reproduce the error. 
Please check the [gist](https://colab.sandbox.google.com/gist/penumallas1261/23ec27f69222caae07508d953332c8fe/untitled0.ipynb) here and feel free to close the issue if it's resolved.
Thanks!

**Comment by pmocz** [comment]
Yes, with more recent versions of jax, e.g. 0.8.0, the runtimes on CPU seem more comparable for this problem (I tried out the colab above on cpus, and tried some bigger matrix sizes too). I will close this issue, and open a new one if I spot something.

Thanks! @penumallas1261

**Comment by penumallas1261** [comment]
Internal reproducer submitted in cl/830262079.
